test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+      authorizationIds = payments.map(payment => createBytesString(payment.authorizationId, BYTES16_LENGTH));
+      accountAddresses = payments.map(payment => payment.account.address);
+      expectedClearedBalances = payments.map(payment => payment.amount);
+      expectedUnclearedBalances = payments.map(() => 0);
+      await setUpContractsForPayments(payments);
+      await addAccountToCardPaymentProcessorWhitelist(admin);
+      await makePayments(payments);
+    });
+
+    it("Is reverted if the contract is paused", async () => {
+      await proveTx(cardPaymentProcessor.setPauser(deployer.address));
+      await proveTx(cardPaymentProcessor.pause());
+      await expect(cardPaymentProcessor.connect(admin).clearPayments(
+        authorizationIds
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_CONTRACT_IS_PAUSED);
+    });
+
+    it("Is reverted if the caller is not whitelisted", async () => {
+      await expect(cardPaymentProcessor.connect(deployer).clearPayments(
+        authorizationIds
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_ACCOUNT_IS_NOT_WHITELISTED);
+    });
+
+    it("Is reverted if the payment authorization IDs array is empty", async () => {
+      await expect(cardPaymentProcessor.connect(admin).clearPayments(
+        []
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_INPUT_ARRAY_OF_AUTHORIZATION_IDS_IS_EMPTY);
+    });
+
+    it("Is reverted if one of the payment authorization IDs is zero", async () => {
+      await expect(cardPaymentProcessor.connect(admin).clearPayments(
+        [authorizationIds[0], ZERO_BYTES16_STRING]
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_IS_ZERO);
+    });
+
+    it("Is reverted if one of the payments with provided authorization IDs does not exist", async () => {
+      await expect(cardPaymentProcessor.connect(admin).clearPayments(
+        [
+          authorizationIds[0],
+          createBytesString(payments[payments.length - 1].authorizationId + 1, BYTES16_LENGTH)
+        ]
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_DOES_NOT_EXIST);
+    });
+
+    it("Is reverted if one of the payments has been already cleared", async () => {
+      await proveTx(cardPaymentProcessor.connect(admin).clearPayment(
+        authorizationIds[0]
+      ));
+      await expect(cardPaymentProcessor.connect(admin).clearPayments(
+        authorizationIds
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_IS_CLEARED);
+    });
+
+    it("Does not transfer tokens, emits the correct events, changes the state properly", async () => {
+      await checkCardPaymentProcessorState(payments);
+      await expect(cardPaymentProcessor.connect(admin).clearPayments(
+        authorizationIds
+      )).to.changeTokenBalances(
+        brlcMock,
+        [cardPaymentProcessor, ...accountAddresses],
+        [0, ...accountAddresses.map(() => 0)]
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "ClearPayment"
+      ).withArgs(
+        authorizationIds[0],
```

Comment:
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
          authorizationIds[0],
```